OpenRA.Game/Map/Map.cs

```diff
@@ -1170,12 +1167,27 @@ public IEnumerable<CPos> FindTilesInAnnulus(CPos center, int minRange, int maxRa
 				foreach (var offset in Grid.TilesByDistance[i])
 				{
 					var t = offset + center;
-					if (valid(t))
+					if (!validate || valid(t))
 						yield return t;
 				}
 			}
 		}
 
+		// Both ranges are inclusive because everything that calls it is designed for maxRange being inclusive:
+		// it rounds the actual distance up to the next integer so that this call
+		// will return any cells that intersect with the requested range circle.
+		// The returned positions are sorted by distance from the center.
+		public IEnumerable<CPos> FindTilesInAnnulus(CPos center, int minRange, int maxRange, bool allowOutsideBounds = false)
+		{
+			return FindTilesInAnnulusInternal(center, minRange, maxRange, allowOutsideBounds, validate: true);
+		}
+
+		// This includes all tiles inside of an annulus, even if they are outside bounds
+		public IEnumerable<CPos> FindAllTilesInAnnulus(CPos center, int minRange, int maxRange)
```

Comment:
This would be a reasonable solution if we only cared about the gen1 mods, but is bogus for the gen2 mods where the cells are not in a regular grid that can be extrapolated outside the map bounds: you may be able to define those cells int terms of an x,y position, but you can't do anything useful such as working out where on the screen they should be drawn.

We can't really avoid hacky code for the chronosphere (the current OpenRA gameplay behaviour was defined back when the engine looked *very* different), but it is far better for those hacks to live in the mod code, where the limitations and assumptions are well defined, than in the engine code where they are not.